.vscode/tasks.json

```diff
@@ -62,50 +81,79 @@
+            "type": "shell",
+            "command":"docker exec -it ${config:container_name} bash -c 'speculos --model ${input:model} build/${input:model}/bin/app.elf'",
+            "group": {
+                "kind": "build",
+                "isDefault": true
+            },
+            "problemMatcher": []
+        },
+        {
+            "label": "Run functional tests",
             "type": "shell",
-            "command": "python /path/to/speculos/speculos.py ${workspaceFolder}/bin/app.elf --ontop --sdk 1.6 --apdu-port 9999 --button-port 42000 --automation-port 43000",
+            "command": "docker exec -it ${config:container_name} bash -c 'pytest tests/ --tb=short -v --device ${input:model}'",
             "group": {
                 "kind": "build",
                 "isDefault": true
             },
-            "dependsOn": [
-                "make debug"
-            ],
             "problemMatcher": []
         },
         {
-            "label": "[debug] run Speculos",
+            "label": "Install tests requirements",
             "type": "shell",
-            "command": "python /path/to/speculos/speculos.py -d ${workspaceFolder}/bin/app.elf --ontop --sdk 1.6 --apdu-port 9999 --button-port 42000 --automation-port 43000",
+            "command":"docker exec -it ${config:container_name} bash -c 'pip3 install -r tests/requirements.txt'",
             "group": {
                 "kind": "build",
                 "isDefault": true
             },
-            "dependsOn": [
-                "make debug"
-            ],
             "problemMatcher": []
         },
         {
-            "label": "run tests",
+            "label": "Load app on device (Linux)",
             "type": "shell",
-            "command": "cd tests && pytest",
+            "command": "if [ ! -f '/etc/udev/rules.d/${config:linux_udev_ledgerblue_rule_file}' ]; then sudo cp .vscode/${config:linux_udev_ledgerblue_rule_file} /etc/udev/rules.d/ && sudo udevadm control --reload-rules && sudo udevadm trigger; fi && docker exec -it ${config:container_name} bash -c 'export BOLOS_SDK=${input:sdk} && make load'",
             "group": {
                 "kind": "build",
                 "isDefault": true
             },
             "problemMatcher": []
         },
         {
-            "label": "kill Speculos",
+            "label": "Load app on device (macOS)",
             "type": "shell",
-            "command": "pkill -f speculos.py",
+            // Checks that virtual env is installed, otherwise installs it. Then installs ledgerblue in a virtualenv and side loads the app APDU file.
+            "command": "cmd=\"[ -d 'ledger' ] || python3 -m virtualenv ledger && source ledger/bin/activate && python3 -m pip install ledgerblue && python3 -m ledgerblue.runScript --scp --fileName bin/app.apdu --elfFile bin/app.elf\"; if python3 -m virtualenv --version >/dev/null 2>&1; then eval \"${cmd}\"; else python3 -m pip install virtualenv && eval \"${cmd}\"; fi",
```

Comment:
This will work once https://github.com/LedgerHQ/blue-loader-python/pull/108 is merge and deployed on PyPI

Reply:
Deployed. 0.1.46 contains necessary updates.